pages/instructorFileEditor/instructorFileEditorClient.js

```diff
@@ -1,22 +1,38 @@
 /* eslint-env browser,jquery */
 /* global ace */
-window.InstructorFileEditor = function(uuid, options) {
-    const elementId = '#file-editor-' + uuid;
-    this.element = $(elementId);
+/* global CryptoJS */
+window.InstructorFileEditor = function(options) {
+    this.element = $(`#${options.elementId}`);
     if (!this.element) {
-        throw new Error('Instructor file editor element ' + elementId + ' was not found!');
+        throw new Error(`Instructor file editor element ${options.elementId} was not found!`);
     }
 
+    this.origHash = options.origHash;
+    this.saveElement = $(`#${options.saveElementId}`);
     this.inputElement = this.element.find('input[name=file_edit_contents]');
     this.editorElement = this.element.find('.editor');
-    this.editor = ace.edit(this.editorElement.get(0));
-    this.editor.setTheme('ace/theme/chrome');
-    this.editor.getSession().setUseWrapMode(true);
-    this.editor.setShowPrintMargin(false);
-    this.editor.getSession().on('change', this.syncFileToHiddenInput.bind(this));
+    this.editor = ace.edit(this.editorElement.get(0), {
+        minLines: 10,
+        maxLines: Infinity,
+        autoScrollEditorIntoView: true,
+        wrap: true,
+        showPrintMargin: false,
+        mode: options.aceMode,
+        readOnly: options.readOnly,
+    });
 
-    if (options.aceMode) {
-        this.editor.getSession().setMode(options.aceMode);
+    if (options.altElementId) {
+        this.editor.setReadOnly(true);
+        this.altElement = $(`#${options.altElementId}`);
+        this.chooseElement = this.element.find('button[id=choose]');
```

Comment:
Choosing my version didn’t work for me on local dev. Steps to reproduce:
1. I opened two windows editing the same file.
2. In window 1 I made change 1 and save-and-synced it.
3. In window 2 I made a different change 2 and clicked save-and-sync.
4. It gave me the “choose” dialog, and I chose “Choose this version (continue editing)”.
5. It gave me the editor with change 2 still present.
6. I clicked save-and-sync.
7. It gave me the same error and “choose” dialog as in step 4.

Instead, I expected that after step 6 the save-and-sync would succeed, overwriting change 1 with change 2.

Reply:
Fixed by replacing `origHash` with `diskHash` on "Choose my version".